modules.rb

```diff
@@ -0,0 +1,35 @@
+module Teachers 
```

Comment:
Why a module here?  We also have common attributes and initialize method -- perfect case for inheritance.